subsys/bluetooth/host/l2cap.c

```diff
@@ -2566,6 +2610,101 @@ int bt_l2cap_ecred_chan_connect(struct bt_conn *conn,
 
 	return err;
 }
+
+#if defined(CONFIG_BT_TESTING)
+static struct bt_l2cap_le_chan *l2cap_find_pending_reconf(struct bt_conn *conn)
+{
+	struct bt_l2cap_chan *chan;
+
+	SYS_SLIST_FOR_EACH_CONTAINER(&conn->channels, chan, node) {
+		if (BT_L2CAP_LE_CHAN(chan)->pending_rx_mtu) {
+			return BT_L2CAP_LE_CHAN(chan);
+		}
+	}
+
+	return NULL;
+}
+
+int bt_l2cap_chan_reconfigure(struct bt_l2cap_chan **chan, uint16_t mtu)
+{
+	struct bt_l2cap_ecred_reconf_req *req;
+	struct bt_conn *conn = NULL;
+	struct bt_l2cap_le_chan *ch;
+	struct net_buf *buf;
+	uint8_t ident;
+	int i;
+
+	for (i = 0; i < L2CAP_ECRED_CHAN_MAX; i++) {
+		if (!chan[i]) {
+			break;
+		}
+
+		/* validate that all channels are from same connection */
+		if (conn) {
+			if (conn != chan[i]->conn) {
+				return -EINVAL;
+			}
+		} else {
+			conn = chan[i]->conn;
+		}
+
+		/* validate MTU is not decreased */
+		if (mtu < BT_L2CAP_LE_CHAN(chan[i])->rx.mtu) {
+			return -EINVAL;
+		}
+	}
+
+	if (i == 0) {
+		return -EINVAL;
+	}
+
+	if (!conn) {
+		return -ENOTCONN;
+	}
+
+	if (conn->type != BT_CONN_TYPE_LE) {
+		return -EINVAL;
+	}
+
+	/* allow only 1 request at time */
+	if (l2cap_find_pending_reconf(conn)) {
+		return -EBUSY;
+	}
+
+	ident = get_ident();
+
+	buf = l2cap_create_le_sig_pdu(NULL, BT_L2CAP_ECRED_RECONF_REQ,
+				      ident,
+				      sizeof(*req) + (i * sizeof(uint16_t)));
+	if (!buf) {
+		return -ENOMEM;
+	}
+
+	req = net_buf_add(buf, sizeof(*req));
+	req->mtu = sys_cpu_to_le16(mtu);
+
+	/* MPS shall not be bigger than MTU + BT_L2CAP_SDU_HDR_SIZE
+	 * as the remaining bytes cannot be used.
+	 */
+	req->mps = sys_cpu_to_le16(MIN(mtu + BT_L2CAP_SDU_HDR_SIZE,
+				       BT_L2CAP_RX_MTU));
+
+	while (i--) {
+		ch = BT_L2CAP_LE_CHAN(chan[i]);
+
+		ch->chan.ident = ident;
+		ch->pending_rx_mtu = mtu;
+
+		net_buf_add_le16(buf, ch->rx.cid);
+	};
+
+	/* we use first channel for sending and timeouting */
+	l2cap_chan_send_req(chan[0], buf, L2CAP_CONN_TIMEOUT);
+
+	return 0;
+}
+#endif
```

Comment:
Could you add a `/* CONFIG_BT_TESTING */` comment here since the opening `#if` is quite far above.